LSPS3/README.md

```diff
@@ -0,0 +1,668 @@
+
+The `opening_fee_params` object describes how much the LSP will charge
+for a channel open, and until when the described feerates will be
+considered valid.
+
+`opening_fee_params.base` is the number of millisatoshis that the client
+has to pay for the cahnnel open, regardless of how large the channel is.
+`opening_fee_params.proportional` is a parts-per-million number that
+describes how many millisatoshis to charge for every million millisatoshis
+of channel size the client wants to open.
+
+`opening_fee_params.valid_until` is a datetime up to which the offered
+`opening_fee_params` is valid, and also serves as the timeout for the
+JIT Channel flow.
+
+> **Rationale** A time limit on the offerred feerates is needed as channel
+> opening is an onchain activity, and onchain feerates need to be reflected
+> in the channel opening feerates the LSP charges.
+> The LSP cannot predict future onchain feerates, and thus must estimate
+> how long its current onchain feerate estimates are valid for.
+
+`opening_fee_params.max_idle_time` is a maximum number of blocks where
+the LSP promises to wait for activity to occur on the channel before it
+closes the channel.
+Once an HTLC has successfully fulfilled, in either direction, on the
+subsequent channel, the LSP SHOULD restart the idle time counter for
+that channel.
+
+`opening_fee_params.max_client_to_self_delay` is a maximum number of blocks
+that the client is allowed to set its `to_self_delay` parameter.
+The client-side `to_self_delay` is the delay imposed on the LSP to recover
+the funds in case the LSP has to perform a unilateral close on the channel
+and defines the security of the client against malign or hacked LSPs.
+
+> **Rationale** Both `max_idle_time` and `max_client_to_self_delay` in
+> total impose a maximum limit on how long a client can lock the funds of
+> the LSP without giving the LSP any opportunity to earn from forwards.
+> Both parameters are placed in the `opening_fee_params` object as they
+> are part of the service that the LSP is promising.
+> Although neither parameter is necessary to compute `opening_fee`, they
+> do define what the client is paying for with that fee.
+
+`opening_fee_params.promise` is an arbitrary string that proves to the LSP
+that it has promised a specific `opening_fee_params` with the specific
+`opening_fee_params.base`, `opening_fee_params.proportional`,
+`opening_fee_params.valid_until`, `opening_fee_params.max_idle_time`, and
+`opening_fee_params.max_client_to_self_delay`.
+
+LSPs:
+
+* SHOULD use a cryptographically-verifiable method of generating
+  `opening_fee_params.promise`, such as a MAC of some deterministic
+  serialization of the `opening_fee_params`, then encoded in hexadecimal
+  or other text encoding.
+* MUST ensure the promise is no longer than 512 bytes.
+* SHOULD avoid characters outside of the printable ASCII range.
+* SHOULD avoid characters that would require escapes in JSON format.
```

Comment:
Cannot be a hash, as the promise must NOT be generatable by anyone other than the LSP, else the client can invent its own parameters and convince the LSP to use those parameters.

The unforgeable cryptographic commitment can be a signature (for private-public crypto, for example the code may be structured so that the issuer of this `promise` is a separate component from the code that validates the `promise` supplied by the client) or it can be a MAC (of which there are several variants) or it can be any future cryptographic algorithm that provides the necessary properties: verifiable (not necessarily third-party-verifiable), unforgeable, and committing to a specific message.